### PR TITLE
add example to add support for HA Energy Dashboard

### PR DESCRIPTION
**Pull request by bvansambeek**
This adds an example to add device in the individual devices drop down list in the HA Energy Dashboard.

## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/total_daily_energy.rst [hunk 1]
```diff
@@ -65,6 +65,34 @@ Some sensors such as the :doc:`HLW8012 <hlw8012>` expose their power sensor with
...
+
+    # Example configuration entry
+    sensor:
+      - platform: total_daily_energy
+        name: "Total Daily Energy"
+        power_id: my_power
+        filters:
+            - multiply: 0.001
+        unit_of_measurement: kWh
+        device_class: energy
```
Comment (frenck) [inline components/sensor/total_daily_energy.rst:82]: This is already its default device class, setting it, is, therefore, not needed.

See: <https://github.com/esphome/esphome/blob/fe535939a3b151d1280d2eddf09213f5eb956153/esphome/components/total_daily_energy/sensor.py#L43>

File: components/sensor/total_daily_energy.rst [hunk 2]
```diff
@@ -65,6 +65,34 @@ Some sensors such as the :doc:`HLW8012 <hlw8012>` expose their power sensor with
...
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+      - platform: total_daily_energy
+        name: "Total Daily Energy"
+        power_id: my_power
+        filters:
+            - multiply: 0.001
+        unit_of_measurement: kWh
```
Comment (frenck) [inline components/sensor/total_daily_energy.rst:81]: There is no need to add filters or multiple to convert from `Wh` to `kWh`, Home Assistant will accept both and handle the conversion automatically.

**Review by frenck — COMMENTED** [review]
I wonder how this example is different compared from the original example on the top of the page?

The default example shown in the introduction of this component is already valid and complete for use with Home Assistant's energy dashboard.

**Comment by bvansambeek** [comment]
Those changes came from a different project with the Integration - Riemann sum integral from Home Assistant and I thought it was also required here, but I didn't test it or thought it through... Sorry. Thanks Frenck for pointing this out and keeping the docs clean.

**Pull request closed without merging**